### PR TITLE
feat(clients/java): get workflow outcome

### LINKED ISSUE #3227: I can get the outcome of workflow with java client
**Is your feature request related to a problem? Please describe.**
Extend jave client to support create workflow instance with results.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

* create a workflow instance and get the results (set of variables) when the workflow is completed. 

## Related issues

Related to #2896 
closes #3227 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java [hunk 1]
```diff
@@ -98,5 +99,14 @@
...
+    /**
+     * When this method is called, the response to the command will be received after the workflow
+     * is completed. The response consists of a set of variables.
+     *
+     * @return the builder for this command. Call {@link #send()} to complete the command and send
+     *     it to the broker
+     */
+    FinalCommandStep<WorkflowInstanceResult> withResult();
   }
 }
```
Comment (Zelldon) [inline clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java]:
```suggestion

}

interface CreateWorkflowInstanceWithResultCommandStep1 extends FinalCommandStep<WorkflowInstanceResult>
{
  // here we could add later more methods
}
```

Comment (Zelldon) [reply clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java]: If you add this interface and return this on withResult() we are able to add more methods without breaking the public api.
It is also more consistent to the other interfaces.

File: clients/java/src/test/java/io/zeebe/client/workflow/CreateWorkflowInstanceWithResultTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+public class CreateWorkflowInstanceWithResultTest extends ClientTest {
+
+  @Test
+  public void shouldCreateWorkflowInstanceByWorkflowInstanceKey() {
+    // given
+    gatewayService.onCreateWorkflowInstanceWithResultRequest(123, "testProcess", 12, 32);
+
+    // when
+    final WorkflowInstanceResult response =
```
Comment (Zelldon) [inline clients/java/src/test/java/io/zeebe/client/workflow/CreateWorkflowInstanceWithResultTest.java]: The key point are the variables I think we should test them as well. That we are able to receive them from the response as map, as object etc.

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
I think we need some more tests for the variable stuff the rest looks good

**Comment by Zelldon** [comment]
Could you also please check the sonargate and the failing builds

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 91697bb7086acba78f436fafded51afe14e9d816
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, ci, build**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Review by ChrisKujawa — COMMENTED** [review]
Thanks @deepthidevaki i like the new tests. Do you think we need also a tests for not returning variables which are not on root scope? Not sure if it makes sense actually. So for example variables which are only available at sub process level or you have variables but in sub process you hide them and change the values only at sub process level.

**Comment by deepthidevaki** [comment]
@Zelldon Isn't it more a test for variable scoping? But I will add one.

**Comment by deepthidevaki** [comment]
@Zelldon  I added the test. Pls have a look.

**Review by ChrisKujawa — APPROVED** [review]
Very good thanks :+1:

**Pull request merged**